From 31f331bcfe7f2a29b4e6cc724306fa58003e0acd Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Tue, 6 Jun 2023 13:24:10 +0200
Subject: [PATCH 0415/1695] dts: stm32mp25: add watchdog support on stm32mp2

Add iwdg1 support on stm32mp2 platform.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a38a5c8a4f94b40ecd539e4d4340e604efd5bdd4 dts: stm32mp25: add watchdog support on stm32mp2]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Signed-off-by: Christophe Roullier <christophe.roullier@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/249550
Change-Id: I37ba75f45182bea415ceb4e6b81ce4d07b6272d8
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/310987
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index a5ed5d98c..c1b7d5155 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -254,7 +254,11 @@
 			};
 
 			iwdg1: watchdog@44010000 {
+				compatible = "st,stm32mp1-iwdg";
 				reg = <0x44010000 0x400>;
+				interrupts = <GIC_SPI 2 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&rcc CK_BUS_IWDG1>, <&rcc LSI_CK>;
+				clock-names = "pclk", "lsi";
 				status = "disabled";
 			};
 		};
-- 
2.47.2

